src/v/raft/state_machine.cc

```diff
@@ -36,6 +36,8 @@ ss::future<> state_machine::start() {
 
 void state_machine::set_next(model::offset offset) { _next = offset; }
 
+ss::future<> state_machine::truncation_detected() { return ss::now(); }
```

Comment:
Makes sense to at least print a warning that we are skipping entries in the default implementation.

Reply:
added logging there